### ISSUE TITLE
Another false negative on EICAR Anti-Malware Testfile

### ISSUE THREAD

**Issue by st-way** [Maikuolan/phpMussel#224]
### phpMussel version used (if known, please also include the exact commit hash).
3.0.1

### Signatures version (if known).
All installed, latest from Git

### PHP SAPI used
litespeed

### PHP version installed
7.2.0

### Operating system used
Linux 3.10.0-962.3.2.lve1.5.36.el7.x86_64

---
## Issue
Hello ! 

Scanning the standard EICAR text file from the eicar website results in a "No problem detected". Test files included with phpMussel are correctly detected. This does not seems to be a access rights problem, as I gave the eicar.txt file the same rights than the test files included in test folder.

**Edit note:** if I rename eicar.txt to eicar.com and un-blacklist .com extension, it is detected as Chameleon Attack. What I don't understand is why nothing is detected when extension is .txt ? My users are uploading TXT, PDF and PPTX files, I need to check these does not contains any kind of malwares inside.

### Expected behavior
Detection of EICAR

### Actual behavior
`string(190) "Wed, 14 Oct 2020 21:13:56 +0200 Started. → Checking "home/[...]/html/eicar.txt". ─→ No problems found. Wed, 14 Oct 2020 21:13:56 +0200 Finished. " `

**Comment by Maikuolan** [comment]
Hi duwayy,

Can you confirm whether there's any file by the name of `phpmussel_regex.db` within your signatures directory?

Cheers. :-)

**Comment by st-way** [comment]
Hi,

Thanks for the quick answer - yes this file is inside the signatures directory

**Comment by st-way** [comment]
Hi Caleb,

Did you have time to investigate on this issue?

And by the way, thanks for maintaining this project... my webhost refused to let me access their clamav service

**Comment by Maikuolan** [comment]
Hi duwayy,

> Did you have time to investigate on this issue?

Sorry for the delayed response. I think I might've uncovered a strange bug in the code, which I haven't quite figured out entirely yet, since I'm seeing some inconsistent results during testing. Working on it currently. I'll reply again once I've figured it out (hopefully a little later tonight, or failing that, early tomorrow morning).

**Comment by Maikuolan** [comment]
> since I'm seeing some inconsistent results during testing.

Okay.. Something slightly annoying, but I'm glad I figured out what was happening, I suppose. The reason I was seeing inconsistent results with my intended bug-fix is that the anti-virus at one of my test machines, for whatever reason, decided to drop some of my exception rules (..sigh), which meant that, although I was able to successfully download eicar onto that machine (in order to test the bug and the intended bug-fix), and able to scan it in CLI (where the bug-fix was working properly), when trying to scan it via standard upload mechanisms, my machine's anti-virus was intercepting it before phpMussel could see it (which made it appear to me as if my intended bug-fix wasn't working properly). Restored my old exceptions.. suddenly everything looks fine again.

Anyway.. Still a bug that was in need of being fixed, but a far, far less complicated one now than it had appeared to me at first glance. And now done ([`v1`](https://github.com/phpMussel/phpMussel/commit/766421818df5dc840bef589aa7ccff723bcaf0db)+[`v2`](https://github.com/phpMussel/phpMussel/commit/6b872483356cf3d6296cab41854dcd8ef973a149)+[`v3Core`](https://github.com/phpMussel/Core/commit/2a7a0a5d4e505f5c75c663b29fd8714c2ca83345)). :-)
